src/buffer.rs

```diff
@@ -86,24 +86,12 @@ pub trait Pixel: Copy + Clone {
     /// that the slice is long enough to present panics if the pixel is used later on.
     fn from_slice_mut(slice: &mut [Self::Subpixel]) -> &mut Self;
 
-    /// Convert this pixel to RGB
-    fn to_rgb(&self) -> Rgb<Self::Subpixel>;
-
     /// Convert this pixel to RGB with an alpha channel
     fn to_rgba(&self) -> Rgba<Self::Subpixel>;
```

Comment:
As long as `to_rgba` is required on `Pixel` it seems to still hold that its color space *must* be interpreted as `sRGB`. But then the other methods, except `invert`, could also be default implemented utilizing this one. Why remove them outright instead?

Reply:
I was trying to remove as much as I could without breaking too much other code in the crate. Perhaps it would be better to step back and consider what generic functionality we can provide on pixels without baking in assumptions about color spaces and so forth. For instance, all of the map/apply variations are rather questionable without knowing the color space

Reply:
That was contrary to my thoughts, `map` and `apply` were the most reasonable in my eye. They do not themselves enforce any particular interpretation. Sure, the caller can not currently find out the color space through `image` but if it is known than the function arguments can correctly interpret the channels.

Reply:
Don't the channels have to be either L or RGB (plus an optional alpha) for that to work? I don't think it could work for things like HSV or xyY since different channels have different meanings but the `map` and `apply` functions would run the same closure on each?

Reply:
I guess one view is that `Pixel` should just be a wrapper around a `&[Subpixel]` plus an associated `ColorType` and `ColorSpace`. It would then fall on the user to properly handle each possible pair they cared about. There would be convenience functions to operate on channels but they'd have no higher level understanding of what they meant.

The alternative is to discourage users from interacting with pixels that way and instead provide higher level functionality so that generic code could work while remaining oblivious to the ColorType and ColorSpace of the pixel. It isn't really obvious to me what this would look like, but seems like the preferable option if feasible

Reply:
> I guess one view is that Pixel should just be a wrapper around a &[Subpixel] plus an associated ColorType and ColorSpace...The alternative is to discourage users from interacting with pixels that way and instead provide higher level functionality

Those may not be mutually exclusive alternatives, but levels of abstraction in the API. Pixels are just buffers with color types and spaces as you say, then higher level pixel traits provide higher level functionality where available. So if a user just needs (strawmen) `Blendable`, `Invertible`, `HasChannel<Channel::Luminance>` or `InterpretableAs<Channel::Red, ColorSpace::sRGB>`, they can construct `fn<A: Invertible + HasChannel<Channel::Luminance>, B: Blendable<A> + InterpretableAs<Channel::Red, ColorSpace::sRGB>> myFancyFilter(a: &A, b: &mut B)`, etc.

If they need something more format-specific not captured in those higher level traits provided by `image` or `imageproc` or other crates, then they can operate on `Pixel`s directly and match/bound the color types and spaces they need to, creating new higher level traits if they wish, e.g., `trait Darken: Pixel {...} impl<P: Pixel<ColorType=RGB, ColorSpace=sRGB>> Darken for P { ... }`.

So while I agree with @HeroicKatora that it's not an entirely coherent change to remove some of these conversions while leaving others, I can see the point in at least shrinking the exposed API since what's exposed is not really coherent as it is, with color spaces and types a bit abused.